geo/src/algorithm/line_locate_point.rs

```diff
@@ -39,58 +41,69 @@ pub trait LineLocatePoint<T, Rhs> {
 
 impl<T> LineLocatePoint<T, Point<T>> for Line<T>
 where
-    T: CoordinateType + Float + Zero + One,
+    T: CoordinateType + Float,
 {
-    type Output = T;
+    type Output = Option<T>;
     type Rhs = Point<T>;
 
     fn line_locate_point(&self, p: &Self::Rhs) -> Self::Output {
-        let sp = [p.x() - self.start.x, p.y() - self.start.y];
-        let v = [self.end.x - self.start.x, self.end.y - self.start.y];
-        let v_sq = v[0] * v[0] + v[1] * v[1];
+        // let $s$ be the starting point of the line, and $v$ its
```

Comment:
Your `$` caused me to learn about KaTex. Nice! I didn't realize rustdoc supported something like that.